src/we_aes_gcm.c

```diff
@@ -356,153 +489,23 @@ static int we_aes_gcm_cipher(EVP_CIPHER_CTX *ctx, unsigned char *out,
         ret = -1;
     }
 
-    decryptFinal = (!aes->enc && len == 0 && in == NULL);
-
-    if ((ret == 1) && aes->tls) {
-        ret = we_aes_gcm_tls_cipher(aes, out, in, len);
-    }
-    else if ((ret == 1) && (out == NULL) & (len != 0)) {
-        WOLFENGINE_MSG(WE_LOG_CIPHER, "Resizing stored AAD and appending "
-                       "data, len = %d", (int)len);
-        /* Resize stored AAD and append new data. */
-        p = (unsigned char*)OPENSSL_realloc(aes->aad, aes->aadLen + (int)len);
-        if (p == NULL) {
-            WOLFENGINE_ERROR_FUNC_NULL(WE_LOG_CIPHER, "OPENSSL_realloc", p);
-            ret = -1;
+    if (ret == 1) {
+        if (aes->tls == 1) {
+            ret = we_aes_gcm_tls_cipher(aes, out, in, len);
         }
         else {
-            /* Copy in new data after existing data. */
-            aes->aad = p;
-            XMEMCPY(aes->aad + aes->aadLen, in, len);
-            aes->aadLen += len;
-            ret = (int)len;
-        }
-    }
-    /* Length may be zero for cases with AAD data only (GMAC) */
-    else if ((ret == 1) && (out != NULL) && (in != NULL || aes->aadLen > 0) &&
-             !decryptFinal) {
-        if (!aes->ivSet) {
-            /* Set extern IV. */
-            WOLFENGINE_MSG(WE_LOG_CIPHER, "Setting external IV");
-            rc = wc_AesGcmSetExtIV(&aes->aes, aes->iv, aes->ivLen);
-            if (rc != 0) {
-                WOLFENGINE_ERROR_FUNC(WE_LOG_CIPHER, "wc_AesGcmSetExtIV", rc);
-                ret = -1;
+            if (out == NULL && len != 0 && in != NULL) {
```

Comment:
Looks like you can make this:
    else if (...) {
    }
    else if (...) {
    }
    else {
    }